thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1362,15 +1388,33 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                 keyManager = activity.createPartKeyManager(partNo, copy);
                 // NB: potentially translation per part could be different if dealing with superkeys
                 setupTranslation(partNo, selected, *keyManager);
+                (*keyManagersCreated)[selected] = true;
             }
             processRows(processing, partNo, keyManager);
         }
+        virtual void getSubFileStats(std::vector<OwnedPtr<CRuntimeStatisticCollection>> & subFileStats) override
+        {
+            for (size_t i=0; i<parts.size(); i++)
+            {
+                if (!(*keyManagersCreated)[i])
```

Comment:
@ghalliday - that's something I missed, I thought there'd be a barrier with a standard memory order, not just effecting or of other atomic operations. Is there no way to ensure that atomic test is not reordered at all?

Reply:
I was wrong.  The atomic write will guarantee that all previous writes are committed, and the read will prevent any reads being migrated before it.
It might be worth using explicit store/load memory order parameters, but in this case the inefficiency of sequential ordering  is unlikely to matter (and doesn't really affect intel).

Reply:
phew, ok that's good to confirm, it sounded that way form my scanning of the docs (a short while ago).
Prior to that I assumed the standard memory ordering of atomics guaranteed it didn't move around reads..
If it did, I suspect (and we were on compilers/architecture where it mattered), my guess was that we'd have other similar code that was affected